reladomographql/src/main/java/com/gs/reladomo/graphql/SDLGenerator.java

```diff
@@ -0,0 +1,319 @@
+/*
+ Copyright 2019 Goldman Sachs.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+
+package com.gs.reladomo.graphql;
+
+import com.gs.fw.common.mithra.MithraManagerProvider;
+import com.gs.fw.common.mithra.attribute.*;
+import com.gs.fw.common.mithra.finder.AbstractRelatedFinder;
+import com.gs.fw.common.mithra.finder.Mapper;
+import com.gs.fw.common.mithra.finder.RelatedFinder;
+import com.gs.fw.common.mithra.transaction.MithraTransactionalResource;
+import com.gs.fw.common.mithra.util.MithraRuntimeCacheController;
+import org.eclipse.collections.impl.set.mutable.UnifiedSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+public class SDLGenerator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger (SDLGenerator.class);
+    private static boolean withComments = false;
+
+    public void generate (final String filename)
+    {
+        try (final Writer out = new FileWriter (filename))
+        {
+            if (withComments) out.write ("# generated on " + LocalDateTime.now () + "\n");
+            this.generate (out);
+
+        } catch (final Exception e)
+        {
+            throw new RuntimeException ("failed to write into " + new File (filename).getAbsolutePath (), e);
+        }
+
+        LOGGER.info ("The complete schema was written to " + new File (filename).getAbsolutePath ());
+    }
+
+    protected void generate (final Writer out) throws IOException
+    {
+        this.writeMutationBlock (out);
+
+        final Set<String> declaredTypes = this.writeQueryBlock (out);
+
+        for (final MithraRuntimeCacheController controller : MithraManagerProvider.getMithraManager ().getRuntimeCacheControllerSet ())
+        {
+            String name = controller.getClassName ();
+            name = StringKit.decapitalize (name, name.lastIndexOf ('.') + 1);
+            name = StringKit.capitalize (name);
+            final RelatedFinder finder = controller.getFinderInstance ();

```

Comment:
Is this the way to iterate through Reladomo metadata at runtime? I was looking for something similar.

Reply:
Metadata is available for classes that are configured via `MithraManagerProvider.getMithraManager ().getRuntimeCacheControllerSet ()`. However, the rest of the code is not quite right. The controller class has a method for getting the metadata: 
```java
public ReladomoClassMetaData getMetaData()
```

`ReladomoClassMetaData` has all sorts of meta data methods as well access to the Finder.

For unconfigured classes that are in the classpath, you can still get the meta data as long as you know the classname, via `ReladomoClassMetaData.from*` methods.

Reply:
Thank you!

Reply:
will update my code. somehow I was not aware of this.

Reply:
BTW, I have implemented the criteria with the Polish Notation of ops along with binary AND operations.
(https://github.com/vborisoff/reladomo/blob/graphql-real/reladomographql/docs/queries.md#Filters) 
I will consider this commend addressed.